rustls-cert-gen/src/cert/mod.rs

```diff
@@ -0,0 +1,44 @@
+use rcgen::{Certificate, CertificateParams, Error};
+use std::{fs::File, io, path::Path};
+
+mod params;
+pub use params::BuildParams;
+
+mod ca;
+mod entity;
+mod signature;
+
+pub trait CertParams {
+	fn params(&self) -> &CertificateParams;
+}
+
+#[derive(Debug, Clone)]
+/// Struct to represent a Serialized Cert/Key pair
+pub struct SerializedEntity {
+	pub cert_pem: String,
+	pub key_pem: String,
+}
+
+impl SerializedEntity {
+	pub fn write(&self, dir: &Path, name: &str) -> Result<(), io::Error> {
+		use std::io::Write;
+
+		let key_path = dir.join(format!("{name}.key.pem"));
+		let mut key_out = File::create(key_path)?;
+		write!(key_out, "{}", &self.cert_pem)?;
```

Comment:
I think this is a bug:
```suggestion
		write!(key_out, "{}", &self.key_pem)?;
```

Maybe there should be a unit test that parses the serialized PEM files and does something with them?